digitalearthau/events.py

```diff
@@ -0,0 +1,116 @@
+import os
+import socket
+import uuid
+from enum import Enum, unique
+
+import datetime
+from typing import NamedTuple, Optional, List
+
+from digitalearthau import pbs
+
+
+class NodeMessage(NamedTuple):
+    hostname: str
+    pid: int
+
+    # An optional id to identify a single instance within a host, such as a worker. (pid is not unique)
+    runtime_id: Optional[uuid.UUID] = None
+
+    def current_node(self) -> 'NodeMessage':
```

Comment:
Should be a `@classmethod`? Helps to mark it as an alternative constructor.

Or is it supposed to be called from an instance?

Reply:
Yep, it should. It's not used anymore in the current code so I may be better to remove it entirely.